flow360/component/volume_mesh.py

```diff
@@ -415,18 +421,76 @@ def _submit_from_surface(self):
         log.info(f"VolumeMesh successfully submitted: {mesh.short_description()}")
         return mesh
 
+    def _compress_and_upload_chunks(
+        self,
+        upload_id: str,
+        mesh: VolumeMesh,
+        remote_file_name: str,
+        max_workers: int = 50,
+        chunk_length: int = 25 * 1024 * 1024,
+    ):
+        uploaded_parts = []  # Initialize an empty list to store the uploaded parts
+        min_upload_size = 5 * 1024 * 1024
+        with open(self.file_name, "rb") as file:
+            parts = []
+            part_number = 1
+            while True:
+                chunk_data = file.read(chunk_length)
+                if not chunk_data:
+                    break
+                compressed_chunk = bz2.compress(chunk_data)
+                while len(compressed_chunk) < min_upload_size and chunk_data:
+                    chunk_data = file.read(chunk_length)
+                    compressed_chunk += bz2.compress(chunk_data)
+                parts.append((part_number, compressed_chunk))
+                part_number += 1
+                executor = concurrent.futures.ThreadPoolExecutor(max_workers=max_workers)
+                future = executor.submit(
+                    mesh.upload_part, remote_file_name, upload_id, part_number, compressed_chunk
+                )
+                uploaded_parts.append(future)
+
+        concurrent.futures.wait(uploaded_parts)
+
+        uploaded_parts = [future.result() for future in uploaded_parts]
+        mesh.complete_multipart_upload(remote_file_name, upload_id, uploaded_parts)
+
+    def _zstd_compress(self, file_path, compression_level=3):
```

Comment:
1. this method can be moved to a more general location. I would suggest flow360->component->utils.py
2. Add optional parameter output_file_path